crypto/bls/src/impls/blst.rs

```diff
@@ -0,0 +1,273 @@
+    }
+
+    fn verify(&self, pubkey: &blst_core::PublicKey, msg: Hash256) -> bool {
+        self.verify(msg.as_bytes(), DST, &[], pubkey) == BLST_ERROR::BLST_SUCCESS
+    }
+}
+
+/// A wrapper that allows for `PartialEq` and `Clone` impls.
+pub struct BlstAggregateSignature(blst_core::AggregateSignature);
+
+impl Clone for BlstAggregateSignature {
+    fn clone(&self) -> Self {
+        Self(blst_core::AggregateSignature::from_signature(
+            &self.0.to_signature(),
+        ))
+    }
+}
+
+impl PartialEq for BlstAggregateSignature {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.to_signature() == other.0.to_signature()
+    }
+}
+
+impl TAggregateSignature<blst_core::PublicKey, BlstAggregatePublicKey, blst_core::Signature>
+    for BlstAggregateSignature
+{
+    fn infinity() -> Self {
+        blst_core::Signature::from_bytes(&INFINITY_SIGNATURE)
+            .map(|sig| blst_core::AggregateSignature::from_signature(&sig))
+            .map(Self)
+            .expect("should decode infinity signature")
+    }
+
+    fn add_assign(&mut self, other: &blst_core::Signature) {
+        self.0.add_signature(other)
+    }
+
+    fn add_assign_aggregate(&mut self, other: &Self) {
+        self.0.add_aggregate(&other.0)
+    }
+
+    fn serialize(&self) -> [u8; SIGNATURE_BYTES_LEN] {
+        self.0.to_signature().to_bytes()
+    }
+
+    fn deserialize(bytes: &[u8]) -> Result<Self, Error> {
+        blst_core::Signature::from_bytes(bytes)
+            .map_err(Into::into)
+            .map(|sig| blst_core::AggregateSignature::from_signature(&sig))
+            .map(Self)
+    }
+
+    fn fast_aggregate_verify(
+        &self,
+        msg: Hash256,
+        pubkeys: &[&GenericPublicKey<blst_core::PublicKey>],
+    ) -> bool {
+        let pubkeys = pubkeys.iter().map(|pk| pk.point()).collect::<Vec<_>>();
+        let signature = self.0.clone().to_signature();
+        signature.fast_aggregate_verify(msg.as_bytes(), DST, &pubkeys) == BLST_ERROR::BLST_SUCCESS
+    }
+
+    fn aggregate_verify(
+        &self,
+        msgs: &[Hash256],
+        pubkeys: &[&GenericPublicKey<blst_core::PublicKey>],
+    ) -> bool {
+        let pubkeys = pubkeys.iter().map(|pk| pk.point()).collect::<Vec<_>>();
+        let msgs = msgs.iter().map(|hash| hash.as_bytes()).collect::<Vec<_>>();
+        let signature = self.0.clone().to_signature();
+        signature.aggregate_verify(&msgs, DST, &pubkeys) == BLST_ERROR::BLST_SUCCESS
+    }
+}
+
+impl TSecretKey<blst_core::Signature, blst_core::PublicKey> for blst_core::SecretKey {
+    fn random() -> Self {
+        let rng = &mut rand::thread_rng();
+        let ikm: [u8; 32] = rng.gen();
+
+        Self::key_gen(&ikm, &[]).unwrap()
+    }
+
+    fn public_key(&self) -> blst_core::PublicKey {
+        self.sk_to_pk()
+    }
+
+    fn sign(&self, msg: Hash256) -> blst_core::Signature {
+        self.sign(msg.as_bytes(), DST, &[])
+    }
+
+    fn serialize(&self) -> SecretHash {
```

Comment:
The name `SecretHash` could be more descriptive.  Something like `AutoZeroizingSecretKey` perhaps?

Reply:
Good call! I named it `ZeroizeHash` :)